libs/knowledge-store/ragstack_knowledge_store/document_graph.py

```diff
@@ -0,0 +1,114 @@
+from __future__ import annotations
+
+import itertools
+from abc import ABC, abstractmethod
+from typing import Any, Iterable, List, Optional, Union, Iterator, Tuple
+
+from langchain_core.callbacks import CallbackManagerForRetrieverRun
+from langchain_core.documents import Document
+from langchain_core.retrievers import BaseRetriever
+from pydantic import Field
+
+
+def _texts_to_documents(
+    texts: Iterator[str],
+    metadatas: Iterator[dict],
+) -> Iterator[Document]:
+    for (
+        text,
+        metadata,
+    ) in zip(texts, metadatas):
+        yield Document(
+            page_content=text,
+            metadata=metadata,
+        )
+
+
+class DocumentGraph(ABC):
```

Comment:
Comments, etc.

Reply:
Yes, I will add doc everywhere once the API is agreed.